google/cloud/bigquery/job.py

```diff
@@ -2112,10 +2118,20 @@ def from_api_repr(cls, resource, client):
         source_config = _helpers._get_sub_prop(
             config_resource, ["extract", "sourceTable"]
         )
-        dataset = DatasetReference(
-            source_config["projectId"], source_config["datasetId"]
-        )
-        source = dataset.table(source_config["tableId"])
+        if source_config:
+            dataset = DatasetReference(
+                source_config["projectId"], source_config["datasetId"]
+            )
+            source = dataset.table(source_config["tableId"])
+        else:
+            source_config = _helpers._get_sub_prop(
+                config_resource, ["extract", "sourceModel"]
+            )
+            dataset = DatasetReference(
+                source_config["projectId"], source_config["datasetId"]
+            )
+            source = dataset.model(source_config["modelId"])
+
```

Comment:
I'd rather replace L2118-2134 with something like this:
```
        ids = _helpers._get_sub_prop(config_resource, ["extract"])
        for id_key in ["tableId", "modelId"]:
            if id_key in ids:
                dataset = DatasetReference(
                    source_config["projectId"], source_config["datasetId"]
                )
                source = dataset.model(source_config[id_key])
                break
```

Reply:
I think break statement is not good, we should avoid to use as much as possible.

Reply:
It might be necessary in this particular case.